lib/org/chromium/webidl/ParserRunner.js

```diff
@@ -0,0 +1,46 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'ParserRunner',
+  extends: 'foam.box.Runnable',
+
+  requires: ['org.chromium.webidl.PipelineMessage'],
+
+  documentation: 'Runnable Box that parsers IDLFileContents into ASTs.',
+
+  methods: [
+    function run(message) {
+      if (!this.PipelineMessage.isInstance(message))
+        return this.error(new Error("ParserRunner: Run() expects a" +
+            " PipelineMessage object!"));
+      else if (!message.idlFile || !message.parser)
+        return this.error(new Error("ParserRunner: Missing required" +
+            " properties in PipelineMessage!"));
+
+      var file = message.idlFile;
+      var parser = message.parser;
+
+      // Load appropriate parser depending on configuration.
+      var parser = foam.lookup(`org.chromium.webidl.${parser}`).create();
```

Comment:
Please add a `TODO`: we should do something safer than using an injected string for class lookup. Perhaps a helper method would inspect the associated URL and instantiate the appropriate parser.